pkg/jobs/update.go

```diff
@@ -119,7 +119,7 @@ func (j *Job) Update(ctx context.Context, txn *kv.Txn, updateFn UpdateFn) error
 	var payload *jobspb.Payload
 	var progress *jobspb.Progress
 	if err := j.runInTxn(ctx, txn, func(ctx context.Context, txn *kv.Txn) error {
-		stmt := "SELECT status, payload, progress FROM system.jobs WHERE id = $1 FOR UPDATE"
+		stmt := "SELECT status, payload, progress FROM system.jobs WHERE id = $1"
```

Comment:
Would it be worth retaining the `FOR UPDATE` clause for calls to `(*Job).Update` that do intend to update the `system.jobs` row? Can we distinguish those from the cases that are just using this function to read by checking whether `updateFn` is a nil (after replacing no-op functions with nil functions)?

Reply:
I don't think that's the tool we want to use. In almost all cases the job is either updated by its coordinator or from pause or cancel. In those cases, I'd actually prefer a pause or cancel to be able to overwrite the status during a long-running update of a job. The only case I think we want locking is this one right here: https://github.com/cockroachdb/cockroach/blob/0cea9dc28cf5b7855be87b6b38d55d191f1b963f/pkg/sql/row/expr_walker.go#L346

I think it's the only place where we try to modify the job from multiple nodes concurrently during normal interaction. I'm going to do some plumbing to retain the locking in that call.